### ISSUE TITLE
Tenssorrt choose fp32 input instead of fp16 input, enouth fp16 is faster.

### ISSUE THREAD

**Issue by a-little-cat** [NVIDIA/TensorRT#2246]
## Description

I write a tensorrt plugin, use it to inference my model(sorry, it not mine, I can't share it).

This plugin support fp16(hwc8) or fp32(liner) input, and output  fp32(liner).

When I trace the performance, I found tensorrt always choose reformater(fp16->fp32) and fp32 input, instead of fp16 input.

In the cache engine generating, the code never run to the  fp16 kernel.

Between, when I close the fp32 support in supportsFormatCombination function, it works as I expected.

I want to know why tensorrt choose the slower reformat and fp32 input instead of fp16 only.

## Environment

**TensorRT Version**: 7.2
**NVIDIA GPU**:  v100 and pg189
**NVIDIA Driver Version**:  450.51
**CUDA Version**: 11.0
**CUDNN Version**: 8.0.4
**Operating System**: ubuntu18
**Python Version (if applicable)**: 
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if so, version)**: 

## Relevant Files

Sorry I don't own them.

**Comment by a-little-cat** [comment]
This plugin has 15 input tensor:
* 9 is conv output(can be fp16_hwc8 or fp32_liner)
* 6 is constant tensor(only fp32_liner)

and 4 output tensor:
* fp32_liner lnly
Thanks for any reply!

**Comment by zerollzeng** [comment]
Can you share the TRT verbose log? I think the verbose tactic optimizer log will tell you why it select the FP32 inputs(most likely your FP32 implementation is faster than FP16, even with the reformat overhead)

**Comment by a-little-cat** [comment]
Thans for your reply!
I tried to get the tensorrt verbose, but there has some problem.(I use onnxruntime, set log level to nvinfer1::ILogger::Severity::kVERBOSE but didn't work😂)

I will keep trying to get the verbose log to checkout.

Besides, I have checked the performance several times, the reformater+fp32 is certainly slow than fp16.
And debug info tell me, tensorrt never go to the fp16 code to test the speed.

**Comment by zerollzeng** [comment]
> And debug info tell me, tensorrt never go to the fp16 code to test the speed.

Perhaps check supportsFormatCombination()

**Comment by a-little-cat** [comment]
I had checked and print  many times. 
I am sure the supportsFormatCombination() is OK.

If input support fp32 and fp16, the supportsFormatCombination () will input the fp16 and fp32input , I return true bother. But then won't go to the fp16 code(means in the enqueue(), input is always fp32).

I have found a way to ignore this problem temporary.
```c++
if (has_fp16(prop.major, prop.minor))
{
    return cur.type == nvinfer1::DataType::kHALF && cur.format == TensorFormat::kHWC8;
}
return cur.type == nvinfer1::DataType::kFLOAT && cur.format == TensorFormat::kLINEAR;
```

I guess maybe tensort has some limit of optimize, when the model is too large, it will choose default format for some plugin.

Feel free to close the issue, I will try to figure it out and if It is done, I will share it here.

Thanks for your reply!
